lib/dependency-manager-adapters/workspace.js

```diff
@@ -0,0 +1,115 @@
+
+module.exports = CoreObject.extend({
+  init() {
+    this._super.apply(this, arguments);
+    this.run = this.run || require('../utils/run');
+  },
+
+  packageJSON: 'package.json',
+
+  setup(options) {
+    if (!options) {
+      options = {};
+    }
+
+    if (!this.useYarnCommand) {
+      throw new Error('workspaces are currently only supported by Yarn, you must set `useYarn` to true');
+    }
+
+    let packageJSON = JSON.parse(fs.readFileSync(this.packageJSON));
+
+    if (!packageJSON.workspaces || !packageJSON.workspaces.length) {
+      throw new Error('you must define the `workspaces` property in package.json with at least one workspace to use workspaces with ember-try');
+    }
+
+    // workspaces is a list of globs, loop over the list and find
+    // all paths that contain a `package.json` file
+    let workspacePaths = [];
+
+    packageJSON.workspaces.forEach(globPattern => {
+      let matchedWorkspacePaths = glob.sync(globPattern, { cwd: this.cwd })
+        .filter(workspacePath => {
+          let packageJSONPath = path.join(this.cwd, workspacePath, 'package.json');
+          return fs.existsSync(packageJSONPath);
+        });
+
+      workspacePaths.push(...matchedWorkspacePaths);
+    });
+
+
+    this._packageAdapters = workspacePaths.map(workspacePath => {
+      return new NpmAdapter({
+        cwd: workspacePath,
+        run: this.run,
+        managerOptions: this.managerOptions,
+        useYarnCommand: this.useYarnCommand,
+      });
+    });
+
+    return RSVP.all(this._packageAdapters.map(adapter => adapter.setup(options)));
+  },
+
+  changeToDependencySet(depSet) {
+    this._packageAdapters.forEach(adapter => {
+      adapter.applyDependencySet(depSet);
+    });
+
+    return this._install().then(() => {
+      let deps = extend({}, depSet.dependencies || {}, depSet.devDependencies || {});
+      let currentDeps = Object.keys(deps).map((dep) => {
+        return {
+          name: dep,
+          versionExpected: deps[dep],
+          versionSeen: this._findCurrentVersionOf(dep),
+          packageManager: this.useYarnCommand ? 'yarn' : 'npm',
+        };
+      });
+
+      debug('Switched to dependencies: \n', currentDeps);
+
+      return RSVP.Promise.resolve(currentDeps);
+    });
+  },
+
+  cleanup() {
+    return RSVP.all(this._packageAdapters.map(adapter => adapter.cleanup()));
+  },
+
+  _install() {
+    let mgrOptions = this.managerOptions || [];
+
+    debug('Run npm install with options %s', mgrOptions);
+
+    let cmd = this.useYarnCommand ? 'yarn' : 'npm';
+
+    if (this.useYarnCommand) {
+      if (mgrOptions.indexOf('--no-lockfile') === -1) {
+        mgrOptions = mgrOptions.concat(['--no-lockfile']);
+      }
+      // npm warns on incompatible engines
+      // yarn errors, not a good experience
+      if (mgrOptions.indexOf('--ignore-engines') === -1) {
+        mgrOptions = mgrOptions.concat(['--ignore-engines']);
+      }
+    } else if (mgrOptions.indexOf('--no-shrinkwrap') === -1) {
```

Comment:
This part was a little pre-emptive, because NPM is supposedly working on support for workspaces, but given our convo offline about ember-try@v2 I think it'd be better to assume we'll be rearchitecting things before then, and will drop these parts